dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue

```diff
@@ -172,21 +507,167 @@ export default defineComponent({
       try {
         this.postEuTaxonomyDataForFinancialsProcessed = false;
         this.messageCount++;
+        let allFileUploadedSuccessful = true;
+        const documentUploadControllerControllerApi = await new ApiClientProvider(
+          assertDefined(this.getKeycloakPromise)()
+        ).getDocumentUploadController();
         const euTaxonomyDataForFinancialsControllerApi = await new ApiClientProvider(
           assertDefined(this.getKeycloakPromise)()
         ).getEuTaxonomyDataForFinancialsControllerApi();
-        this.postEuTaxonomyDataForFinancialsResponse =
-          await euTaxonomyDataForFinancialsControllerApi.postCompanyAssociatedEuTaxonomyDataForFinancials(
-            this.formInputsModel
+
+        if (this.filesToUpload.length) {
+          for (let index = 0; index < this.filesToUpload.length; index++) {
+            const uploadFileSuccessful = await documentUploadControllerControllerApi.postDocument(
+              this.filesToUpload[index]
+            );
+            if (!uploadFileSuccessful) {
+              allFileUploadedSuccessful = false;
+              break;
+            } else if (uploadFileSuccessful) {
+              this.updatePropertyFilesUploaded(
+                index,
+                "documentId",
+                uploadFileSuccessful.data.documentId,
+                "filesToUpload"
+              );
+            }
+          }
+        }
+        if (allFileUploadedSuccessful) {
+          await this.$nextTick();
+          const formInputsModelToSend = modifyObjectKeys(
+            JSON.parse(JSON.stringify(this.formInputsModel)) as objectType,
+            "send"
           );
-        this.$formkit.reset("createEuTaxonomyForFinancialsForm");
+          this.postEuTaxonomyDataForFinancialsResponse =
+            await euTaxonomyDataForFinancialsControllerApi.postCompanyAssociatedEuTaxonomyDataForFinancials(
+              formInputsModelToSend
+            );
+        }
       } catch (error) {
         this.postEuTaxonomyDataForFinancialsResponse = null;
         console.error(error);
       } finally {
         this.postEuTaxonomyDataForFinancialsProcessed = true;
+        this.confirmedSelectedKPIs = [];
+        this.selectedKPIs = [];
+        this.fiscalYearEnd = "";
+        this.filesToUpload = [];
+        this.uploadFiles = [];
+        this.formInputsModel = {};
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        this.$refs.UploadReports.clearAllNotUploadedFiles();
+        await this.$nextTick();
+        this.$formkit.reset("createEuTaxonomyForFinancialsForm");
```

Comment:
Just a suggestio: You could make the whole method shorter by modularizing this part of the code to something like `resetForm()` or smth smiliar.

Reply:
I will have to touch this method anyways when I will work on the new sub-tasks.

So you don't need to do this modularizing, since I can do it during my work on it.

Reply:
The reset is not needed anymore since a redirect happens after submission.
Therefore no modularizing needed for it.